include/dali/core/tensor_shape.h

```diff
@@ -243,12 +243,20 @@ struct TensorShape<DynamicDimensions>
   /**
    * @brief Return empty shape of specified dimensionality
    */
-  static TensorShape<> empty_shape(int dim) {
+  static TensorShape empty_shape(int dim) {
```

Comment:
Just to clarify, I should be correct in removing "<>" this since the template is specified at the top (struct TensorShape<-1>). For the other variant below, TensorShape itself hasn't been templated (template <int ndim> struct TensorShape), so we have to specity TensorShape<ndim> whenever its used.

This compiles and passes the slicing tests, but doesn't mean its correct (r.e. previous bug).

Reply:
Yeah, removing the template argument works, too - it's even better this way.